watch.go

```diff
@@ -21,36 +20,60 @@ type Event struct {
 	BatchId uint64
 }
 
-// Watcher temporarily stores event information,
+type WatchChan <-chan *Event
+
+type Watcher interface {
+	Watch() WatchChan
+	Close() error
+}
+
+// watcher temporarily stores event information,
 // as it is generated until it is synchronized to DB's watch.
 //
 // If the event is overflow, It will remove the oldest data,
 // even if event hasn't been read yet.
-type Watcher struct {
-	queue eventQueue
-	mu    sync.RWMutex
+type watcher struct {
+	queue      eventQueue
+	mu         sync.RWMutex
+	watchCh    chan *Event
+	flagCh     chan struct{}
+	cancelFunc func()
 }
 
-func NewWatcher(capacity uint64) *Watcher {
-	return &Watcher{
+func newWatcher(capacity uint64, test ...bool) *watcher { // test is only for UT
+	w := &watcher{
 		queue: eventQueue{
 			Events:   make([]*Event, capacity),
 			Capacity: capacity,
 		},
+		watchCh: make(chan *Event, 100),
+		flagCh:  make(chan struct{}, capacity),
```

Comment:
can we use a `sync.Cond` to do the notification? There are two channels with the same capacity, seems weird.